src/Text/Pandoc/Writers/Markdown.hs

```diff
@@ -1045,6 +1045,12 @@ inlineToMarkdown opts (Emph lst) = do
                       then "_" <> contents <> "_"
                       else contents
               else "*" <> contents <> "*"
+inlineToMarkdown _ (Underline []) = return empty
+inlineToMarkdown opts (Underline lst) = do
+  contents <- inlineListToMarkdown opts lst
+  return $ if isEnabled Ext_raw_html opts
+              then "<u>" <> contents <> "</u>"
+              else contents
```

Comment:
Raw HTML is not ideal.  I'm thinking we should render underline in Markdown the way we've been doing already, as a bracketed Span with a class (falling back to raw HTML only when bracketed Spans aren't enabled).  (And the Markdown reader should recognize that syntax as Underline.)  That will give us round-trip behavior without the mess of raw HTML.  What do you think?  Maybe also, when bracketed spans aren't enabled but native spans are, we should use `<span class="">` syntax.

Reply:
I implemented this change in both the reader and writer. I wasn't sure if the class should be `ul` as indicated above or `underline`, so my current approach supports both.